flink-docs/src/main/java/org/apache/flink/docs/configuration/ConfigOptionsDocGenerator.java

```diff
@@ -524,8 +524,19 @@ static String getDescription(OptionWithMetaInfo optionWithMetaInfo) {
             return null;
         }
 
+        Set<String> enumConstantNameSet =
+                Arrays.stream(clazz.getDeclaredFields())
+                        .filter(Field::isEnumConstant)
+                        .filter(
+                                field ->
+                                        !field.isAnnotationPresent(
+                                                Documentation.ExcludeFromDocumentation.class))
+                        .map(Field::getName)
+                        .collect(Collectors.toSet());
+
         InlineElement[] optionDescriptions =
                 Arrays.stream(clazz.getEnumConstants())
+                        .filter(constant -> enumConstantNameSet.contains(((Enum) constant).name()))
```

Comment:
There's no need to store all the non-excluded constants in a set. You can convert a `Field` into a enum constant by calling `field.get(null)`.